src/domain.c

```diff
@@ -508,17 +611,27 @@ enum OperationReturnCode ld_add_entry(LDHandle *handle, const char *name, const
     check_string(name, entry_name, "ld_add_entry");
     check_string(parent, entry_parent, "ld_add_entry");
 
-    TALLOC_CTX *talloc_ctx = talloc_new(NULL);
+    TALLOC_CTX *talloc_ctx = NULL;
+    ld_talloc_new(talloc_ctx, error_exit, NULL);
 
-    const char* dn = talloc_asprintf(talloc_ctx,"%s=%s,%s", prefix, entry_name, entry_parent);
+    const char* dn;
+    ld_talloc_asprintf(dn, error_exit, talloc_ctx,"%s=%s,%s", prefix, entry_name, entry_parent);
 
     LDAPMod **attrs = fill_attributes(entry_attrs, talloc_ctx, LDAP_MOD_ADD);
 
     rc = add(handle->connection_ctx, dn, attrs);
 
-    talloc_free(talloc_ctx);
+    ld_talloc_free(talloc_ctx, error_exit);
 
     return rc;
+
+    error_exit:
+        if (talloc_ctx)
+        {
+            ld_talloc_free(talloc_ctx, error_exit);
```

Comment:
Do not use error_exit mark in error_exit statement. For explanation Look for description of first similar issue at the beginning of the file.